src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -41,6 +46,63 @@ trait ExportEntitiesByType extends FireCloudRequestBuilding {
   implicit val userInfo: UserInfo
   implicit protected val executionContext: ExecutionContext
 
+  import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+
+  /**
+    * Overall approach:
+    * Generate a list of all queries to extract all of the entities
+    * For each of those, generate individual lists
+    * Stream that list back out to the sender.
+    * TODO: Add zip-streaming for set types
+    * TODO: Add filename handling
+    * TODO: Add attributeName handling
+    */
+  def streamEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[Stream[String]] = {
+    val sortField = entityType + "_id"
+    val firstQuery: EntityQuery = EntityQuery(page = 1, pageSize = 1, sortField = sortField, sortDirection = SortDirections.Ascending, filterTerms = None)
+
+    // Get all of the possible queries
+    lazy val pageQueriesFuture: Future[Seq[EntityQuery]] = getQueryResponse(workspaceNamespace, workspaceName, entityType, firstQuery) map {
+      queryResponse =>
+        val pageSize = 500 // TODO: Should this be a config?
+      val pages = queryResponse.resultMetadata.filteredCount/pageSize match {
+        case x if x > Math.floor(x) => (Math.floor(x) + 1).toInt
+        case x => x.toInt
+      }
+        val range = 1 to pages
+        range map {
+          page =>
+            EntityQuery(page = page, pageSize = pageSize, sortField = sortField, sortDirection = SortDirections.Ascending, filterTerms = None)
+        }
+    }
+
+    // Make those queries and generate a nested mess of streams
+    lazy val nestedEntityFutures: Future[Seq[Future[Seq[Entity]]]] = pageQueriesFuture map { querySeq =>
```

Comment:
@rtitle - Yes, this whole section of the code needs a lot of work. I'm pretty sure it's not working the way I really want it to ... I'll likely ask for your help next week :-)